scripts/scil_labelseg.py

```diff
@@ -0,0 +1,155 @@
+website: https://pytorch.org/get-started/locally/
+
+Parts of the implementation are based on or lifted from:
+    SAM-Med3D: https://github.com/uni-medical/SAM-Med3D
+    Multidimensional Positional Encoding: https://github.com/tatp22/multidim-positional-encoding
+
+To cite: Antoine Théberge, Zineb El Yamani, François Rheault, Maxime Descoteaux, Pierre-Marc Jodoin (2025). LabelSeg. ISMRM Workshop on 40 Years of Diffusion: Past, Present & Future Perspectives, Kyoto, Japan.  # noqa
+"""
+
+import argparse
+import nibabel as nib
+import os
+
+from argparse import RawTextHelpFormatter
+
+from scilpy.io.utils import (
+    assert_inputs_exist, assert_outputs_exist, add_overwrite_arg)
+from scilpy.image.volume_operations import resample_volume
+
+from scilpy.ml.labelseg.predict import predict
+from scilpy.ml.labelseg.utils import get_data, get_model, download_weights
+from scilpy.ml.utils import get_device
+
+
+from dipy.utils.optpkg import optional_package
+
+IMPORT_ERROR_MSG = "PyTorch is required to run this script. Please install" + \
+                   " it first. See the official website for more info: " + \
+                   "https://pytorch.org/get-started/locally/"  # noqa
+torch, have_torch, _ = optional_package('torch', trip_msg=IMPORT_ERROR_MSG)
+
+# TODO: Get bundle list from model
+DEFAULT_BUNDLES = ['AF_left', 'AF_right', 'ATR_left', 'ATR_right', 'CA', 'CC_1', 'CC_2', 'CC_3', 'CC_4', 'CC_5', 'CC_6', 'CC_7', 'CG_left', 'CG_right', 'CST_left', 'CST_right', 'FPT_left', 'FPT_right', 'FX_left', 'FX_right', 'ICP_left', 'ICP_right', 'IFO_left', 'IFO_right', 'ILF_left', 'ILF_right', 'MCP', 'MLF_left', 'MLF_right', 'OR_left', 'OR_right', 'POPT_left', 'POPT_right', 'SCP_left', 'SCP_right', 'SLF_III_left', 'SLF_III_right', 'SLF_II_left', 'SLF_II_right', 'SLF_I_left', 'SLF_I_right', 'STR_left', 'STR_right', 'ST_FO_left', 'ST_FO_right', 'ST_OCC_left', 'ST_OCC_right', 'ST_PAR_left', 'ST_PAR_right', 'ST_POSTC_left', 'ST_POSTC_right', 'ST_PREC_left', 'ST_PREC_right', 'ST_PREF_left', 'ST_PREF_right', 'ST_PREM_left', 'ST_PREM_right', 'T_OCC_left', 'T_OCC_right', 'T_PAR_left', 'T_PAR_right', 'T_POSTC_left', 'T_POSTC_right', 'T_PREC_left', 'T_PREC_right', 'T_PREF_left', 'T_PREF_right', 'T_PREM_left', 'T_PREM_right', 'UF_left', 'UF_right']  # noqa E501
+
+DEFAULT_CKPT = os.path.join('checkpoints', 'labelsegnet.ckpt')
+
+
+def _build_arg_parser(parser):
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=RawTextHelpFormatter)
+
+    parser.add_argument('fodf', type=str,
+                        help='fODF input')
+    parser.add_argument('wm', type=str,
+                        help='WM input')
+    parser.add_argument('out_prefix', type=str,
+                        help='Output destination and file prefix.')
+    parser.add_argument('--mask', type=str, default=None,
+                        help='Output destination and file prefix for '
+                             'binary mask output.')
```

Comment:
I don't get why we need a mask here.

Reply:
The method takes in as input a whole-brain WM mask and can output a bundle-wise binary mask. I'll make this clearer in the doc.

Reply:
Actually I could also just remove the option (I've kept it since we get it "for free" as the model outputs one) as it can be obtained by binarizing the label map.